packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/src/ScmTypeFactory.cs

```diff
@@ -23,6 +23,24 @@ public class ScmTypeFactory : TypeFactory
 
         public virtual CSharpType TokenCredentialType => throw new NotImplementedException("Token credential is not supported in Scm libraries yet");
 
+        public virtual CSharpType? ClientUriBuilderBaseType => null;
```

Comment:
where we would use this?

Reply:
The generation code for Azure needs it: https://github.com/Azure/azure-sdk-for-net/pull/46530/files#diff-35b8ffef55a67d289e0a5b6b51d70226759746e7fc2cd76c7ef3e3d81f0c71c9R16